code401challenges/src/test/java/code401challenges/utilities/AnimalShelterTest.java

```diff
@@ -0,0 +1,31 @@
+package code401challenges.utilities;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class AnimalShelterTest {
+    @Test
+    public void testEnqueueDequeue() {
+        AnimalShelter<String> actual = new AnimalShelter<>();
+        actual.enqueue("cat");
+        actual.enqueue("cat");
+        actual.enqueue("dog");
+
+        assertEquals("cat", actual.dequeue("cat"));
+        assertEquals("cat", actual.dequeue("cat"));
+        assertEquals("dog", actual.dequeue("dog"));
```

Comment:
You should be able to put these assert statements in any order and have the test pass.